nighthawk/source/common/utility.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <string>
+
+namespace Nighthawk {
+
+namespace PlatformUtils {
+uint32_t determineCpuCoresWithAffinity();
+}
+
+class Uri {
+public:
+  static Uri Parse(std::string uri) { return Uri(uri); }
+
+  const std::string& host_and_port() const { return host_and_port_; }
+  const std::string& host_without_port() const { return host_without_port_; }
+  const std::string& path() const { return path_; }
+  uint64_t port() const { return port_; }
+  const std::string& scheme() const { return scheme_; }
+
+  bool isValid() {
+    return (scheme_ == "http" || scheme_ == "https") && (port_ > 0 && port_ <= 65535);
+  }
+
+private:
+  Uri(std::string uri);
+
+  // TODO(oschaaf): username, password, etc. But we may want to look at
+  // pulling in a mature uri parser.
+  std::string host_and_port_;
+  std::string host_without_port_;
+  std::string path_;
+  uint64_t port_;
+  std::string scheme_;
+  bool is_https_;
```

Comment:
It seems there is some redundancy in the strings above and things like schme_/is_https_. Seems like this is probably motivated by the needs of external code. Would it be possible to minimize this interface a bit?

Reply:
I removed `is_https_`; However, I put in `host_with_portname` and `host_without_portname` on purpose, to hopefully create awareness of that a distinct choice should be made here when using it. (In reaction to me messing up earlier by putting in the host without the portname in the `host:` header). Considering that, WDYT?